server/src/test/java/io/spine/examples/chatspn/server/message/MessageTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.chatspn.server.message;
+
+import io.spine.examples.chatspn.ChatId;
+import io.spine.examples.chatspn.MessageId;
+import io.spine.examples.chatspn.message.Message;
+import io.spine.examples.chatspn.message.command.PostMessage;
+import io.spine.examples.chatspn.message.event.MessagePosted;
+import io.spine.examples.chatspn.server.ChatsContext;
+import io.spine.server.BoundedContextBuilder;
+import io.spine.testing.core.given.GivenUserId;
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static io.spine.testing.TestValues.randomString;
+
+@DisplayName("`Message` should")
+class MessageTest extends ContextAwareTest {
+
+    @Override
+    protected BoundedContextBuilder contextBuilder() {
+        return ChatsContext.newBuilder();
+    }
+
+    @Test
+    @DisplayName("allow posting and emit the `MessagePosted` event")
+    void creation() {
+        PostMessage command = PostMessage
+                .newBuilder()
+                .setId(MessageId.generate())
+                .setChat(ChatId.generate())
+                .setUser(GivenUserId.generated())
+                .setContent(randomString())
+                .vBuild();
+
+        context().receivesCommand(command);
+
+        MessagePosted expectedEvent = MessagePosted
+                .newBuilder()
+                .setId(command.getId())
+                .setChat(command.getChat())
+                .setUser(command.getUser())
+                .setContent(command.getContent())
+                .build();
```

Comment:
I think, it's copy-paste of your another test.

It should be `vBuild()`. Otherwise, why would you define all the constraints in the Proto definitions.

Same below.